lib/trento_web/views/v1/sap_system_view.ex

```diff
@@ -152,32 +161,43 @@ defmodule TrentoWeb.V1.SapSystemView do
   defp add_system_replication_status_to_secondary_instance(
          %{database_instances: database_instances} = sap_system
        ) do
-    system_replication_status =
-      Enum.find_value(database_instances, fn
-        %{
-          system_replication: "Primary",
-          system_replication_status: system_replication_status
-        } ->
-          system_replication_status
-
-        _ ->
-          false
-      end)
+    system_replication_status = get_system_replication_status(database_instances)
 
     database_instances =
-      Enum.map(database_instances, fn
-        %{
-          system_replication: "Secondary"
-        } = instance ->
-          %{instance | system_replication_status: system_replication_status}
+      Enum.map(
+        database_instances,
+        &map_system_replication_status_to_secondary(&1, system_replication_status)
+      )
 
-        %{system_replication: "Primary"} = instance ->
-          %{instance | system_replication_status: ""}
+    Map.put(sap_system, :database_instances, database_instances)
+  end
 
-        instance ->
-          instance
-      end)
+  defp add_system_replication_status(instance, database_instances) do
+    system_replication_status = get_system_replication_status(database_instances)
+    map_system_replication_status_to_secondary(instance, system_replication_status)
+  end
 
-    Map.put(sap_system, :database_instances, database_instances)
+  defp get_system_replication_status(database_instances) do
+    Enum.find_value(database_instances, fn
+      %{
+        system_replication: "Primary",
+        system_replication_status: system_replication_status
+      } ->
+        system_replication_status
+
+      _ ->
+        false
+    end)
   end
+
+  defp map_system_replication_status_to_secondary(
+         %{system_replication: "Secondary"} = instance,
+         system_replication_status
+       ),
+       do: %{instance | system_replication_status: system_replication_status}
+
+  defp map_system_replication_status_to_secondary(%{system_replication: "Primary"} = instance, _),
```

Comment:
Just a question: why do primary instances get set to `""` here? To me the name `map_system_replication_status_to_secondary()` makes me think that primary instances won't be affected by this function

Reply:
The code is exactly as it was before.
The secondary state pill is only sent to `Secondary` instances, that's why it is set to an empty string (as it was before).
About the naming, well, this is pattern matching technique, so the 1st and 2nd functions must have the same name, and at the end, the functions sets the status only to secondary instances